packages/server/src/user/user.service.ts

```diff
@@ -191,5 +216,29 @@ export class UserService {
     }
   }
 
+  /**
+   * Update user's role
+   *
+   * @param id uuid of the user as string
+   * @param roleToEdit role needs to edit in number representation, refer to `role.enum.ts`
+   * @param addRole `true` for add new role to user, `false` for remove role from user
+   */
+  async updateUserRole(id: string, roleToEdit: number, addRole: boolean): Promise<void> {
+    const userToUpdate = await this.findUserById(id);
+
+    // Add a role: role OR role_to_add
+    // Remove a role: role XOR role_to_remove
+    const role = addRole ? userToUpdate.role | roleToEdit : userToUpdate.role ^ roleToEdit;
+
+    await this.prisma.user.update({
+      where: {
+        id: userToUpdate.id
+      },
+      data: {
+        role: role
```

Comment:
NIT: can be simplfied
```suggestion
        role
```